parser/internal/parservX/parser.go

```diff
@@ -818,6 +831,141 @@ func (p *parser) parse() (ast.Expression, error) {
 	return result, nil
 }
 
+func (p *parser) parseFunctionDefinitions() error {
+	for {
+		tok, lit, _ := p.scanIgnoreWhitespace()
+		if tok == ast.Name && lit == "def" {
+			function, err := p.parseFunctionDefinition()
+			if err != nil {
+				return err
+			}
+			p.functions[function.GetID()] = function
```

Comment:
It's not possible for `p.functions` to be nil. I think it makes sense for this to then return a parse error:

```suggestion
			if p.functions == nil {
				return parse error with "functions are not supported"
			}
			p.functions[function.GetID()] = function
```

Reply:
something isn't working right if we make that check. Something else is wrong then, because we never _get to_ parse the functions, at least not in the tests.

Reply:
ok, I found where in the tests it wasn't being initialized => coming as nil. should be fine now.